### PR TITLE
fix: Permission error when opening marketplace projects (marketplace projects need to add any view permission link)

### PR DESCRIPTION
**Pull request by istarkov**
## Description

- [x] - Todo cursor and other styles issues

ref #3446

We now display the marketplace project link only if it has a shared link with view permission. If there is no shared link, we show disabled buttons with a tooltip.

image

image

## Steps for reproduction

Open marketplace, see one has link the other is not

image

Check that link is working in incognito

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/features/sidebar-left/panels/marketplace/about.tsx [hunk 1]
```diff
@@ -75,21 +78,39 @@ export const About = ({
...
+              hasAuthToken
+                ? builderUrl({
+                    projectId: item.projectId,
+                    origin: location.origin,
+                    authToken: item.authorizationToken,
+                  })
+                : undefined
+            }
+            target="_blank"
+            data-state={hasAuthToken ? undefined : "disabled"}
```
Comment (TrySound) [inline apps/builder/app/builder/features/sidebar-left/panels/marketplace/about.tsx:109]: aria-disabled=true

File: apps/builder/app/builder/features/sidebar-left/panels/marketplace/templates.tsx [hunk 2]
```diff
@@ -169,17 +175,33 @@ export const Templates = ({
...
+            hasAuthToken
+              ? undefined
+              : 'The project does not have a shared link with "View" permission.'
+          }
         >
-          <ExternalLinkIcon />
-        </Link>
+          <Link
+            css={{
+              "&[data-state=disabled], &[data-state=disabled]:visited": {
```
Comment (TrySound) [inline apps/builder/app/builder/features/sidebar-left/panels/marketplace/templates.tsx:187]: Does :visited overwrites color? Afaik it always use explicitly defined one on unvisited selector.

File: apps/builder/app/shared/context.server.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import {
   getTokenPlanFeatures,
   getUserPlanFeatures,
 } from "./db/user-plan-features.server";
-import { getAllApprovedProjectIds } from "./marketplace/db.server";
+// import { getAllApprovedProjectIds } from "./marketplace/db.server";
```
Comment (TrySound) [inline apps/builder/app/shared/context.server.ts:10]: remove comment?

File: packages/design-system/src/components/button.tsx [hunk 4]
```diff
@@ -96,10 +96,11 @@ const perColorStyle = (variant: ButtonColor) => ({
           ),
   },
 
-  "&:disabled:not([data-state=pending]), &[data-state=disabled]": {
-    background: theme.colors.backgroundButtonDisabled,
-    color: theme.colors.foregroundDisabled,
-  },
+  "&:disabled:not([data-state=pending]), &[data-state=disabled], &[data-state=disabled]:visited, &[data-state=disabled]:hover":
```
Comment (TrySound) [inline packages/design-system/src/components/button.tsx:99]: same there is perfectly valid aria-disabled=true

File: apps/builder/app/builder/features/sidebar-left/panels/marketplace/templates.tsx (hunk 2 above)
Comment (istarkov) [reply apps/builder/app/builder/features/sidebar-left/panels/marketplace/templates.tsx]: moved to Link style

File: packages/design-system/src/components/button.tsx (hunk 4 above)
Comment (istarkov) [reply packages/design-system/src/components/button.tsx]: Moved

**Review by kof — APPROVED** [review]

**Review by TrySound — APPROVED** [review]

**Pull request merged**